### PR TITLE
refactored spa.Compare to use Product net

### PR DESCRIPTION
**Pull request by Seanny123**
As mentioned in #798 .

### CODE REVIEW

File: nengo/spa/compare.py [hunk 1]
```diff
@@ -52,16 +47,15 @@ def __init__(self, dimensions, vocab=None, neurons_per_multiply=200,
...
         with self:
             nengo.Connection(self.inputA,
-                             self.compare.input[::2], synapse=None)
+                             self.product.A, synapse=None)
             nengo.Connection(self.inputB,
-                             self.compare.input[1::2], synapse=None)
-            self.compare.add_output('product', lambda x: x[0] * x[1])
+                             self.product.B, synapse=None)
 
     def on_add(self, spa):
```
Comment (xchoo) [inline nengo/spa/compare.py]: Is this function necessary now? Originally it was done on add because a vocabulary was being used (I believe)

Comment (Seanny123) [inline nengo/spa/compare.py]: Fixed.

File: nengo/spa/compare.py [hunk 2]
```diff
@@ -18,14 +18,12 @@ class Compare(Module):
...
-        Effective radius for the multiplication.  The actual radius will
+    input_magnitude : float
+        Effective input_magnitude for the multiplication.  The actual input_magnitude will
         be this value times sqrt(2)
-    direct : boolean
-        Whether or not to use direct mode for the neurons
     """
     def __init__(self, dimensions, vocab=None, neurons_per_multiply=200,
-                 output_scaling=1.0, radius=1.0, direct=False,
+                 output_scaling=1.0, input_magnitude=1.0,
```
Comment (xchoo) [inline nengo/spa/compare.py]: Is the output_scaling factor necessary? @tcstewar thoughts? (I removed it in my version of the compare network).

Comment (Seanny123) [inline nengo/spa/compare.py]: From my personal experience, I agree that it isn't necessary.

Comment (tcstewar) [inline nengo/spa/compare.py]: Yeah, I'd get rid out output_scaling.

Comment (Seanny123) [inline nengo/spa/compare.py]: Done. Is that the last change? Should I squash my commits and mark to merge?

**Comment by Seanny123** [comment]
I've addressed all the comments made so far. Should I mark this ready to merge?

**Comment by tbekolay** [comment]
@xchoo can you give this a final lookover? If it looks good I'll merge.

**Comment by xchoo** [comment]
lgtm!

**Comment by tcstewar** [comment]
Yup, looks good to me too.  :)

**Pull request merged**